src/verify_cert.rs

```diff
@@ -985,20 +941,30 @@ mod tests {
         )
     }
 
-    #[test]
-    #[cfg(feature = "alloc")]
-    fn longest_allowed_path() {
-        assert_eq!(build_linear_chain(1), Ok(()));
-        assert_eq!(build_linear_chain(2), Ok(()));
-        assert_eq!(build_linear_chain(3), Ok(()));
-        assert_eq!(build_linear_chain(4), Ok(()));
-        assert_eq!(build_linear_chain(5), Ok(()));
-        assert_eq!(build_linear_chain(6), Ok(()));
+    fn make_issuer(org_name: String) -> rcgen::Certificate {
+        let mut ca_params = rcgen::CertificateParams::new(Vec::new());
+        ca_params
+            .distinguished_name
+            .push(rcgen::DnType::OrganizationName, org_name);
+        ca_params.is_ca = rcgen::IsCa::Ca(rcgen::BasicConstraints::Unconstrained);
+        ca_params.key_usages = vec![
+            rcgen::KeyUsagePurpose::KeyCertSign,
+            rcgen::KeyUsagePurpose::DigitalSignature,
+            rcgen::KeyUsagePurpose::CrlSign,
+        ];
+        ca_params.alg = &rcgen::PKCS_ECDSA_P256_SHA256;
+        rcgen::Certificate::from_params(ca_params).unwrap()
     }
 
-    #[test]
-    #[cfg(feature = "alloc")]
-    fn path_too_long() {
-        assert_eq!(build_linear_chain(7), Err(Error::MaximumPathDepthExceeded));
+    fn make_end_entity(issuer: &rcgen::Certificate) -> CertificateDer<'_> {
```

Comment:
Style nit: this actually returns `CertificateDer<'static>`, right? I'd make that explicit -- I feel like we usually reserve the elided lifetime for input lifetimes.

Reply:
Good idea. Thanks.